[SooYap] iP

SooYap opened the pull request:
RainyBot

“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

RainyBot frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
FAST SUPER FAST to use
All you need to do is,

download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/276).
double-click it.
add your tasks.
let it manage your tasks for you 😉
And it is FREE!

Features:

1. Managing tasks
2. Managing deadlines (coming soon)
3. Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rainy.java:
Change name
Add automatic messages

Commit:
Rainy.java:
Add Scanner

Commit:
Rainy.java: Add echo messages and Bye response

Commit:
TaskTracker.java: Add new TaskTracker class

Commit:
TaskTracker.java: Include methods for adding and previewing tasks

Commit:
Rainy.java: Add Constructor and TaskTracker attribute

Commit:
Rainy.java: Add functionality to add tasks and view tasks

Commit:
TaskTracker.java: Change variable name due to bug

Commit:
Task.java: Add Task class

Commit:
TaskTracker.java: Changed type of taskList from String to Task

Commit:
Task.java: Change to getName for Task retrieval

Commit:
TaskTracker: Add message when viewing tasks

Commit:
Task.java: Add mark and unmark methods

Commit:
TaskTracker.java: Add mark and unmark methods

Commit:
Rainy.java: Provide implementations for the mark and unmark functions

Commit:
TaskTracker.java: Add Error messages for invalid task numbers

Commit:
Deadline.java: Add Deadline class

Commit:
Event.java: Add Event class

Commit:
ToDo.java: Add ToDo class

Commit:
TaskTracker.java: Add individual methods when updating taskList for ToDo, Deadline, and Event

Commit:
Deadline.java: Override toString method

Commit:
Event.java: Override toString method

Commit:
ToDo.java: Override toString method

Commit:
Rainy.java: Add popup message when task is added

Commit:
TaskTracker.java: Add notice for number of tasks

Commit:
Rainy.java: remove newline

Commit:
EXPECTED.TXT: add all expected outputs

Commit:
input.txt: add all input lines

Commit:
runtest.sh: add correct directory using javac and java commands

Commit:
InvalidIndexException.java: add InvalidIndexException class

Commit:
TaskTracker.java: throw InvalidIndexException when user tries to mark/unmark an invalid task

Commit:
Rainy.java: add error messages for invalid ToDo and Deadline user inputs

Commit:
Rainy.java: add error messages for invalid user input for event and error message for not specifying child class of Task

Commit:
InvalidMarkAndUnmarkException: create InvalidMarkAndUnmarkException class

Commit:
Task.java: throw InvalidMarkAndUnmarkException when user marks or unmarks incorrectly

Commit:
TaskTracker.java: declare InvalidMarkandUnmarkException under mark and unmark methods

Commit:
Rainy.java: declare InvalidMarkAndUnmarkException under main method

Commit:
TaskTracker.java: use collections to represent taskList and add delete method

Commit:
Rainy.java: add implementation to delete task from tm

Commit:
TaskTracker.java: throws InvalidIndexException if user enters an invalid index

Commit:
Instructions.java: add Instructions enum

Commit:
Rainy.java: change user instructions to enum and change if else statements to switch

Commit:
Rainy.java: Add support for saving data using File and FileWriter objects

TanSooYap committed:
Rainy.java: Add support for File Object

TanSooYap committed:
Merge branch 'branch-Level-7'

TanSooYap committed:
Deadline.java: Add Date and Time parser for user input

TanSooYap committed:
Event.java: Add Date and Time parser for user inputs

TanSooYap committed:
Rainy.java: Change input processing for Event and Deadline

TanSooYap committed:
Rainy.java: Add sort functionality to sort task list by date in increasing order

TanSooYap committed:
Merge branch 'branch-Level-8'

TanSooYap committed:
UI.java: Create UI class for chatbot outputs

TanSooYap committed:
Rainy.java: Transfer chat outputs to UI class

TanSooYap committed:
Storage.java: Add Storage class to load and save tasks to file

TanSooYap committed:
TaskTracker.java: Add toggleReceivedInputs method

TanSooYap committed:
Rainy.java: Move copy and write files function to Storage class

TanSooYap committed:
Parser.java: Add Parser class for processing user inputs

TanSooYap committed:
input.txt: Update test cases

TanSooYap committed:
Rainy.java: Move input processing to Parser class

TanSooYap committed:
Rainy.java: Shift file write location

TanSooYap committed:
Merge branch 'branch-A-MoreOOP'

TanSooYap committed:
Rainy.java: Add packages

TanSooYap committed:
Merge branch 'branch-A-Packages'

TanSooYap committed:
runtest.sh: Change file path to relative and account for packages

TanSooYap committed:
Merge branch 'add-gradle-support'

TanSooYap committed:
build.gradle: Add checkstyle plugin and application file path for running program.

TanSooYap committed:
Deadline.java: Add space character in output

TanSooYap committed:
TaskTest.java: Create new TaskTest class to test the mark and unmark methods

TanSooYap committed:
TaskTrackerTest.java: Add new TaskTrackerTest class to test the mark and unmark methods

TanSooYap committed:
Merge branch 'branch-A-JUnit'

TanSooYap committed:
TaskTrackerTest.java: Add tests for other non-trivial methods

TanSooYap committed:
build.gradle: Change file path

TanSooYap committed:
Merge branch 'branch-A-Jar'

TanSooYap committed:
Instructions.java: Add JavaDoc comments to every method

TanSooYap committed:
Parser.java: Add JavaDoc comments to every method

TanSooYap committed:
Storage.java: Add JavaDoc comments to every method

TanSooYap committed:
UI.java: Add JavaDoc comments to every method

TanSooYap committed:
InvalidIndexException.java: Add JavaDoc comments to every method

TanSooYap committed:
InvalidMarkAndUnmarkException.java: Add JavaDoc comments to every method

TanSooYap committed:
Deadline.java: Add JavaDoc comments to every method

TanSooYap committed:
Event.java: Add JavaDoc comments to every method

TanSooYap committed:
Task.java: Add JavaDoc comments to every method

TanSooYap committed:
TaskTracker.java: Add JavaDoc comments to every method

TanSooYap committed:
ToDo.java: Add JavaDoc comments to every method

TanSooYap committed:
Rainy.java: Add JavaDoc comments to every method

TanSooYap committed:
Merge branch 'branch-A-JavaDoc'

TanSooYap committed:
Rainy.java: Change code according to coding standards

TanSooYap committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/rainy/database/Rainy.java
#	src/main/java/rainy/database/UI.java
#	src/main/java/rainy/tasks/Task.java
#	src/main/java/rainy/tasks/TaskTracker.java

Resolution: Keep both

TanSooYap committed:
Rainy.java: Add find method

TanSooYap committed:
Rainy.java: Edit the find method to make it functional

TanSooYap committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/rainy/database/Rainy.java
#	src/main/java/rainy/tasks/Event.java
#	src/main/java/rainy/tasks/TaskTracker.java

Resolution: Accept branch-Level-9 find method and JavaDoc comments and CodingStandard from master branch

TanSooYap committed:
TaskTracker.java: Add JavaDoc comments for findTask method

TanSooYap committed:
checkstyle.xml: Add checkstyle and suppression files

hansneddyanto commented on src/main/java/rainy/database/Rainy.java:
```diff
@@ -0,0 +1,166 @@
+
+        // Start processing user inputs
+        while(instruction != Instructions.BYE) {
+            switch (instruction) {
+                case LIST:
+                    System.out.println(tm.getList());
+                    break;
+
+                case MARK:
+                    if (count != -1) {
+                        tm.markDone(count - 1);
+                    } else {
+                        ui.noCategoryDeclared();
+                    }
+                    if (tm.getCounter() > 0) {
+                        File f = new File("src/main/java/rainy.txt");
```
Note there is no indentation for case clauses.

SooYap replied:
I've changed it, thank you!!

hansneddyanto commented on src/main/java/rainy/tasks/Task.java:
```diff
@@ -0,0 +1,77 @@
+ * name, date, and a <code>UI</code> object to display chatbot messages.
+ */
+public class Task implements Comparable<Task> {
+    private boolean isDone;
```
I like how you make sure that boolean variable name has 'is' in front (i.e. using 'isDone' instead of 'done')

hansneddyanto commented on src/main/java/rainy/rainyexceptions/InvalidIndexException.java:
```diff
@@ -0,0 +1,17 @@
+package rainy.rainyexceptions;
+import rainy.database.*;
+import rainy.tasks.*;
```
Imported classes should always be listed explicitly.

hansneddyanto commented on src/main/java/rainy/database/Storage.java:
```diff
@@ -0,0 +1,90 @@
+
+    }
+
+    /**
+     * Takes in a <code>File</code> object and reads the data.
+     * @param newFile                        Represents the file object to be read.
+     * @return                               This method copies the previously established tasks to a newly created <code>TaskTracker</code>
+     *                                       object and returns it to be used by the Rainy chatbot.
+     * @throws InvalidIndexException         Thrown by <code>TaskManager</code> object when user provides a non-existent task number.
+     * @throws InvalidMarkAndUnmarkException Thrown by <code>Task</code> object when user wants to mark a marked tasked or unmark an unmarked task.
+     */
```
I like how you use JavaDocs in most of your public methods. It is complete with param, return and even throw

hansneddyanto reviewed (commented):
Overall LGTM! Good job!

TanSooYap committed:
Rainy.java: Edit code to align with proper checkstyle

TanSooYap committed:
Merge branch 'branch-A-CheckStyle'

TanSooYap committed:
Main.java: Start elementary JavaFX

TanSooYap committed:
Rainy.java: Add JavaFX edits

chuajunyu commented on src/main/java/rainy/database/Parser.java:
```diff
@@ -0,0 +1,101 @@
+     * @throws NumberFormatException  This method catches the exception when a <code>String</code> cannot be
+     *                                converted into an <code>Integer</code>.
+     */
+    public void firstInput(String userInput) {
```
Perhaps the naming of this method could be reworded into something more like a verb instead that could capture what the method is doing more accurately?

Maybe something like `validateUserInput`?

chuajunyu commented on src/main/java/rainy/database/Parser.java:
```diff
@@ -0,0 +1,101 @@
+     *                                   this method catches an IllegalArgumentException and assigns it
+     *                                   Instructions.INVALID.
+     */
+    public Instructions enumOperator(String newMessage) {
```
Perhaps the naming of this method could be reworded into a verb that more accurately describes what the method is doing too?

Maybe something along the lines of `convertStringToInstruction` ?

SooYap replied:
Thanks I'll keep that in mind!!

chuajunyu commented on src/main/java/rainy/tasks/TaskTracker.java:
```diff
@@ -0,0 +1,221 @@
+ *
+ */
+public class TaskTracker {
+    private ArrayList<Task> taskList;
```
Perhaps you could rename this to `tasks` to since this is a collection of Tasks and it should be plural?

chuajunyu commented on src/main/java/rainy/tasks/TaskTracker.java:
```diff
@@ -0,0 +1,221 @@
+public class TaskTracker {
+    private ArrayList<Task> taskList;
+    private int counter;
+    private boolean receivedInputs;
```
Perhaps you could rename this such that it sounds more like a boolean?

maybe `isInputReceived` or `hasReceivedInput`?

SooYap replied:
Thank you for the suggestion :)

chuajunyu commented on src/test/java/rainy/tasks/TaskTest.java:
```diff
@@ -0,0 +1,36 @@
+public class TaskTest {
+
+    @Test
+    public void markTest() throws InvalidMarkAndUnmarkException {
```
Perhaps you can consider using the following three part format `featureUnderTest_testScenario_expectedBehavior()` to name your test cases in a more descriptive manner?

chuajunyu reviewed (commented):
Overall, I think code quality is good with regards to having descriptive and apt naming. Some improvements would be naming methods as verbs, as some of them sound like nouns. There are some small coding standard violations such as plural variables or boolean variables, but they are small and can be fixed easily. No major errors overall, good job!

TanSooYap committed:
Rainy.java: Add JavaFX GUI for chatbot

TanSooYap committed:
Merge branch 'branch-Level-10'

TanSooYap committed:
Storage.java: Add streams to method

TanSooYap committed:
TaskTracker.java: Add streams to find and get methods

TanSooYap committed:
Rainy.java: Add assert statements

TanSooYap committed:
Storage.java: Fix bugged feature regarding file overwriting for tasks with indices 9 and above

TanSooYap committed:
Command.java: Refactor code based on SLAP principle

TanSooYap committed:
Command.java: Create Command classes

TanSooYap committed:
ByeCommand: Refactor bye implementation

TanSooYap committed:
Command.java: Create Command implementation

TanSooYap committed:
DeadlineCommand.java: Refactor Deadline Command implementation

TanSooYap committed:
Delete.java: Refactor Delete Command implementation

TanSooYap committed:
EventCommand.java: Refactor Event Command implementation

TanSooYap committed:
FindCommand.java: Refactor Find Command implementation

TanSooYap committed:
List.java: Refactor List Command implementation

TanSooYap committed:
Mark.java: Refactor Mark Command implementation

TanSooYap committed:
SortCommand.java: Refactor Sort Command implementation

TanSooYap committed:
ToDoCommand.java: Refactor ToDo Command implementation

TanSooYap committed:
Unmark.java: Refactor Unmark Command implementation

TanSooYap committed:
Command.java: Change class to an abstract one

All Common Subclasses implement the getResponse() method.

Hence, it makes sense for the Command Class to be abstract, containing an abstract method getResponse().

Furthermore, although some Subclasses do not require for their getResponse() methods to return anything,
I have made it so that they will artificially return a TaskTracker object, to fit into the constraints of the abstract method.

SooYap committed:
Merge pull request #1 from SooYap/branch-A-Assertions

branch-A-Assertions

SooYap committed:
Merge branch 'master' into branch-A-Streams

TanSooYap committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/rainy/database/Rainy.java
#	src/main/java/rainy/tasks/TaskTracker.java

TanSooYap committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/rainy/database/Rainy.java

TanSooYap committed:
Merge branch 'branch-A-Streams' of https://github.com/SooYap/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/rainy/commands/ByeCommand.java
#	src/main/java/rainy/commands/Command.java
#	src/main/java/rainy/commands/List.java
#	src/main/java/rainy/commands/Mark.java
#	src/main/java/rainy/commands/Unmark.java
#	src/main/java/rainy/database/Rainy.java

TanSooYap committed:
TaskTracker.java: Fix bugs

SooYap committed:
Merge pull request #4 from SooYap/branch-A-Streams

branch-A-Streams

SooYap committed:
Merge pull request #3 from SooYap/branch-A-CodeQuality

branch-A-CodeQuality

TanSooYap committed:
Storage.java: Abstract out the copyPreviousFiles method into several lower-level methods

Within the copyPreviousFiles lie the functionality to read different lines of the saved text file. Depending on what the kind of task it is (ToDo, Deadline, Event), the method provides different ways of processing the line.

Abstracting the processes into lower-level methods serves to trim the code and provide better readability.

Further, this method also processes tasks based on whether they have been marked as done. This function has been abstracted into a separate, lower-level method to improve code readability.

Due to the method's implementation of streams and AtomicIntegers, the previous code was cluttered and disorganized. This has been improved upon.

TanSooYap committed:
Rainy.java: Add update implementation

TanSooYap committed:
Rainy.java: Start implementation of the update function

TanSooYap committed:
Rainy.java: Improve update functionality

This increment is decidedly important as it gives users the ability
to edit their previous task additions.

Since there are three different types of tasks, I had to separately
implement the functionality for each different task.

For ToDo, it is the most simple since it only takes in a string, it
accepts any kind of text input.

For Deadline, users are required to specify whether they want to
change the name of the task or the date of the task.

Similarly, for event, users need to specify which parameter they are
changing, the name, date, or time of the event.

All other functionalities of the chatbot remain intact.

TanSooYap committed:
Merge branch 'branch-C-Update'

TanSooYap committed:
Rainy.java: Replace magic literals with static constants

TanSooYap committed:
Rainy.java: Add exception handling for file overwriting

TanSooYap committed:
Remove wrong file

TanSooYap committed:
Add correct file

TanSooYap committed:
Rainy.java: Edit code to align with coding standard.

TanSooYap committed:
EventCommand.java: Edit GUI output for error messages

TanSooYap committed:
Update README.md

TanSooYap committed:
Add new format for displaying warning messages

TanSooYap committed:
Add functionality to create folder to store data file

TanSooYap committed:
Update method return type

TanSooYap committed:
Add error handling for invalid commands

TanSooYap committed:
Add functionality to create folder to store data file

TanSooYap committed:
Update code to align with coding standard

TanSooYap committed:
Include comments showing inspiration for GUI solution

TanSooYap committed:
Fix indexing problems when printing tasks

SooYap committed:
Update README.md
